src/test/java/io/jenkins/plugins/junit/storage/database/DatabaseTestResultStorageTest.java

```diff
@@ -60,202 +78,269 @@
 
 public class DatabaseTestResultStorageTest {
 
-    public static final String TEST_IMAGE = "postgres:16-alpine";
+    private static final String TEST_IMAGE = "postgres:16-alpine";
+    private static final String BUILD_PIPELINE =
+            "node('remote') {\n" +
+            "    writeFile file: 'x.xml', text: '''<testsuite name='sweet' time='200.0'>\n" +
+            "        <testcase classname='Klazz' name='test1' time='198.0'><error message='failure'/></testcase>\n" +
+            "        <testcase classname='Klazz' name='test2' time='2.0'/>\n" +
+            "        <testcase classname='other.Klazz' name='test3'><skipped message='Not actually run.'/></testcase>\n" +
+            "    </testsuite>'''\n" +
+            "    def s = junit 'x.xml'\n" +
+            "    echo(/summary: fail=$s.failCount skip=$s.skipCount pass=$s.passCount total=$s.totalCount/)\n" +
+            "    writeFile file: 'x.xml', text: '''<testsuite name='supersweet'>\n" +
+            "        <testcase classname='another.Klazz' name='test1'><error message='another failure'/></testcase>\n" +
+            "    </testsuite>'''\n" +
+            "    s = junit 'x.xml'\n" +
+            "    echo(/next summary: fail=$s.failCount skip=$s.skipCount pass=$s.passCount total=$s.totalCount/)\n" +
+            "}\n";
+
+
     @ClassRule
     public static BuildWatcher buildWatcher = new BuildWatcher();
 
     @Rule
-    public JenkinsRule r = new JenkinsRule();
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @Before
+    public void setup() {
+        Logger logger = Logger.getLogger(DatabaseTestResultStorage.class.getName());
+        logger.setLevel(Level.INFO);
+
+        ConsoleHandler consoleHandler = new ConsoleHandler();
+        consoleHandler.setLevel(Level.INFO);
+        logger.addHandler(consoleHandler);
+
+        // Remove default handlers to prevent duplicate messages
+        Logger rootLogger = Logger.getLogger("");
+        Handler[] handlers = rootLogger.getHandlers();
+        for (Handler handler : handlers) {
+            rootLogger.removeHandler(handler);
+        }
+        rootLogger.addHandler(consoleHandler);
```

Comment:
Not sure exactly what you were doing here, seems you're setting logging to INFO?

This would probably simplify what you are trying to do: https://github.com/jenkinsci/jenkins-test-harness/blob/master/src/main/java/org/jvnet/hudson/test/LoggerRule.java

Reply:
Yep, that's great. Didn't know about it. I've implemented it.